### PR TITLE
chore: adding tests for edge cases for processCommands & StartPullerAndProcessModels

### PR DESCRIPTION
**Pull request by zhaque44**
**Testing the StartPullerAndProcessModels Function:**

**Scenarios:**
1.) Tested with a valid downloader, commands, and a logger.
2.) Tested with a nil downloader.
3.)  Tested with nil commands and a valid downloader.
4.) Tested with a nil logger.

`TestProcessCommands`
**Scenarios:**
1.) Puller is initialized with an empty command channel, simulating a scenario with no commands. 
2.) Puller is set up with a command channel containing an invalid model spec. The test verifies that the Puller correctly processes the invalid spec, and the opStats map remains empty.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhaque44** [comment]
@alexagriffith @andyi2it Hi all, happy new year. It's my first time contributing, please let me know if there is anything I am missing.

File: pkg/agent/puller_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	logger := zap.NewNop().Sugar()
+	downloader := &Downloader{}
+	puller := Puller{
+		channelMap:  make(map[string]*ModelChannel),
+		completions: make(chan *ModelOp, 4),
+		opStats:     make(map[string]map[OpType]int),
+		waitGroup:   WaitGroupWrapper{sync.WaitGroup{}},
+		Downloader:  downloader,
+		logger:      logger,
+	}
```
Comment (terrytangyuan) [inline pkg/agent/puller_test.go]: These seem a bit repetitive. Perhaps refactoring a bit and run different test cases in a loop?

Comment (zhaque44) [reply pkg/agent/puller_test.go]: hey @terrytangyuan that's actually a great idea. I refactored them to be table-driven test, which cover the same assertions and validations for each of the separate test functions.

**Comment by zhaque44** [comment]
@terrytangyuan @alexagriffith @andyi2it Verify Generated Code / verify-codegen github action fails and it says to run make generate: when i run that make target it says:

hack/update-codegen.sh
chmod: /Users/zubairhaque/go/pkg/mod/k8s.io/code-generator@v0.28.4/generate-groups.sh: No such file or directory
make: *** [generate] Error 1

^ can you guys help me with this?

**Comment by terrytangyuan** [comment]
Install k8s.io/code-generator

**Comment by terrytangyuan** [comment]
For this specific change, you'll only need to add the copyright header

**Comment by zhaque44** [comment]
@terrytangyuan hey thanks alot for your help: i have installed that go package: i have a subsequent error:

```
ERROR: 
	 API rule check failed. Reported violations in file hack/current_violation_exceptions.list differ from known violations in file hack/violation_exceptions.list. 

make: *** [generate] Error 1
```

when i do a `git status`

the modified changes below? so many files have been deleted and updated thats not right? many more files have been deleted besides the one listed below:

	modified:   go.mod
	modified:   go.sum
	deleted:    pkg/apis/serving/v1alpha1/zz_generated.deepcopy.go
	modified:   pkg/apis/serving/v1beta1/openapi_generated.go
	deleted:    pkg/apis/serving/v1beta1/zz_generated.deepcopy.go
	deleted:    pkg/client/clientset/versioned/clientset.go
	deleted:    pkg/client/clientset/versioned/fake/clientset_generated.go
	deleted:    pkg/client/clientset/versioned/fake/doc.go
	deleted:    pkg/client/clientset/versioned/fake/register.go
	deleted:    pkg/client/clientset/versioned/scheme/doc.go
	deleted:    pkg/client/clientset/versioned/scheme/register.go

im kinda worried that this is not right, also seems pretty odd for a simple test coverage change. I am willing to work on this if its worth it for this project

**Comment by terrytangyuan** [comment]
You can manually add copyright header to the new file

**Comment by zhaque44** [comment]
@terrytangyuan thanks again! feel free to review again whenever you have time, no rush
